From acfc27b50dfc80da1d24e14bbb41106b85bf8d77 Mon Sep 17 00:00:00 2001
From: Pat Noziska <patrick.j.noziska@intel.com>
Date: Mon, 6 Jul 2015 16:12:17 -0700
Subject: [PATCH] Fix Coverity Issues in Intel driver modules.

Provides fixes for AndroidKernel issues found by Coverity
scan of 2015/06/25 in Intel modules that are not
inherited from upstream.

Change-Id: I4d0a9ca4b053d55b4c036a5f93a26bfd7de78abf
Signed-off-by: Pat Noziska <patrick.j.noziska@intel.com>

--- linux.orig/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c	2015-07-07 15:30:10.198248760 -0700
+++ linux/drivers/external_drivers/camera/drivers/media/i2c/mt9m114.c	2015-07-07 15:30:58.630248197 -0700
@@ -2225,6 +2225,7 @@
 	if (ret) {
 		v4l2_device_unregister_subdev(&dev->sd);
 		kfree(dev);
+		return ret;
 	}
 #else
 	if (client->dev.platform_data) {
Index: linux/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
===================================================================
--- linux.orig/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c	2015-07-07 15:30:10.198248760 -0700
+++ linux/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c	2015-07-07 15:30:58.630248197 -0700
@@ -5890,7 +5890,7 @@
 			struct atomisp_ae_window *arg)
 {
 	struct atomisp_device *isp = asd->isp;
-	struct v4l2_subdev_selection sel;
+	struct v4l2_subdev_selection sel = { 0 };
 
 	sel.r.left = arg->x_left;
 	sel.r.top = arg->y_top;
Index: linux/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
===================================================================
--- linux.orig/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c	2015-07-07 15:30:10.198248760 -0700
+++ linux/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c	2015-07-07 15:30:58.630248197 -0700
@@ -1739,7 +1739,7 @@
 
 int atomisp_alloc_metadata_output_buf(struct atomisp_sub_device *asd)
 {
-	unsigned int i;
+	int i;
 
 	/* We allocate the cpu-side buffer used for communication with user
 	 * space */
Index: linux/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/dvs/dvs_1.0/ia_css_dvs.host.c
===================================================================
--- linux.orig/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/dvs/dvs_1.0/ia_css_dvs.host.c	2015-07-07 15:30:10.198248760 -0700
+++ linux/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/isp/kernels/dvs/dvs_1.0/ia_css_dvs.host.c	2015-07-07 15:30:58.630248197 -0700
@@ -66,7 +66,7 @@
 	unsigned int uv_flag)
 {
 	unsigned int i, j;
-	gdc_warp_param_mem_t s;
+	gdc_warp_param_mem_t s = { 0 };
 	unsigned int x00, x01, x10, x11,
 		     y00, y01, y10, y11;
 
Index: linux/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_dvs.c
===================================================================
--- linux.orig/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_dvs.c	2015-07-07 15:30:10.198248760 -0700
+++ linux/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_param_dvs.c	2015-07-07 15:30:58.630248197 -0700
@@ -214,8 +214,8 @@
 			(*dvs_6axis_config)->ycoords_uv = NULL;
 		}
 
-		sh_css_free(*dvs_6axis_config);
 		IA_CSS_LEAVE_PRIVATE("dvs_6axis_config %p", (*dvs_6axis_config));
+		sh_css_free(*dvs_6axis_config);
 		*dvs_6axis_config = NULL;
 	}
 }
